azure-devops/azext_devops/dev/common/credential_store.py

```diff
@@ -33,10 +33,14 @@ def set_password(self, key, token):
                 keyring.delete_password(key, self._USERNAME)
```

Comment:
We are not clearing password from file here. But I cannot think of a scenario where this would have any impact so looks ok. I am still mentioning this to think through any scenario which could cause concern.

It could just end up having credentials stored in both file and keyring (steps below) but it should be taken care by logout.

Steps:
1. Keyring dependency is missing and user runs az devops login
2. Keyring issue gets sorted out.
3. User runs az  devops login without running logout.

Reply:
I agree, but i think that's fine since we are giving explicit warning that we are storing credentials in file and asking them to logout later.